rules-reviewed/eap7/eap6/hibernate4.windup.xml

```diff
@@ -604,5 +572,160 @@
+                    </message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#changedmoved-contracts"
+                        title="Hibernate 5 Changed or Moved Contracts" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hibernate4-00034">
+            <when>
+                <javaclass references="org.hibernate.engine.jdbc.spi.JdbcServices">
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Class org.hibernate.engine.jdbc.spi.JdbcServices were changed in Hibernate 5" effort="1" severity="optional">
+                    <message>Hibernate 5 extracted new interface `org.hibernate.engine.jdbc.env.spi.JdbcEnvironment` from `org.hibernate.engine.jdbc.spi.JdbcServices`.</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#changedmoved-contracts"
+                        title="Hibernate 5 Changed or Moved Contracts" />
+                     <link
+                        href="https://docs.jboss.org/hibernate/orm/5.0/javadocs/index.html?org/hibernate/engine/jdbc/env/spi/JdbcEnvironment.html"
+                        title="Hibernate 5 JdbcEnvironment new API" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hibernate4-00035">
+            <when>
+                <javaclass references="org.hibernate.id.Configurable">
+                    <location>IMPLEMENTS_TYPE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Interface org.hibernate.id.Configurable changed method parameter in configure method in Hibernate 5" effort="1" severity="mandatory">
+                    <message>Hibernate 5 changed the signature of `org.hibernate.id.Configurable#configure` method to accept `ServiceRegistry` rather than just `Dialect`</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#changedmoved-contracts"
+                        title="Hibernate 5 Changed or Moved Contracts" />
+                     <link
+                        href="https://docs.jboss.org/hibernate/orm/5.0/javadocs/org/hibernate/id/Configurable.html#configure-org.hibernate.type.Type-java.util.Properties-org.hibernate.service.ServiceRegistry-"
+                        title="Hibernate 5 Configurable API" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hibernate4-00036">
+            <when>
+                <javaclass references="org.hibernate.id.PersistentIdentifierGenerator">
```

Comment:
Should this instead look for implementations of `org.hibernate.id.Configurable`?